fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -2345,11 +2352,12 @@ public Resource doInvoke(FHIROperationContext operationContext, String resourceT
         try {
             Class<? extends Resource> resourceType = null;
             if (resourceTypeName != null) {
-                resourceType = (Class<? extends Resource>) getResourceType(resourceTypeName);
+                resourceType = getResourceType(resourceTypeName);
             }
+            String operationKey = (resourceTypeName == null? operationName : operationName + ":" + resourceTypeName);
```

Comment:
```suggestion
            String operationKey = (resourceTypeName == null ? operationName : operationName + ":" + resourceTypeName);
```
added a space

Reply:
you may not need branching logic here.  why not set ResourceType as Resource.class or Single a specific class for the hash  so you don't need branching logic.

Reply:
"Resource" is used for operations which supports all resource types.  different meaning...

Reply:
right - it was an optimization on the key